File: support/domain/ddd/DomainEntity.cs
using support.domain;
namespace support.domain.ddd{
    /// <summary>
    /// Generic interface for marking domain entities
    /// </summary>
    /// <typeparam name="EID">Generic-Type of the entity identifier</typeparam>
    public interface DomainEntity<EID>:Identifiable<EID>{
        ///<summary>Represents the entity hashcode</summary>
        ///<returns>Integer with the entity hashcode</returns>
        ///
        int GetHashCode();
        
        ///
        ///<summary>Checks if two entities are equal</summary>
        ///<param name="obj">DomainEntity with the entity being compared to the current one</param>
        ///<returns>boolean true if both entities are equal, false if not</returns>
        ///
        bool Equals(object obj);
    }
}

File: MYCA/myca/support/src/main/java/cdiomyc/support/domain/ddd/DomainEntity.java
package cdiomyc.support.domain.ddd;

import cdiomyc.support.domain.Identifiable;

/**
 * Generic Interface for marking domain entities
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 * @param <EID> Generic-Type of the domain entity identifier
 */
public interface DomainEntity<EID> extends Identifiable<EID>{
    /**
     * Returns the hashcode of the domain entity
     * @return Integer with the hash code of the domain entity
     */
    @Override
    int hashCode();
    
    /**
     * Checks if a domain entity is equal to the current one
     * @param otherDomainEntity DomainEntity with the comparing domain entity
     * @return boolean true if both domain entities are equal, false if not
     */
    @Override
    boolean equals(Object otherDomainEntity);
}
